.vale/justfile

```diff
@@ -1,64 +1,37 @@
 COLOR := "\\033[0;34m"
 NO_COLOR := "\\033[0m"
 
-
 # Show all available recipes
 @_default:
     printf "\n{{ COLOR }}# Vale (path: \`.vale/\`)\n"
     printf "===================={{ NO_COLOR }}\n"
     just --list --unsorted
 
-
 # Build a local version of `openverse-vale:local` for testing
 build:
     docker build . -t openverse-vale:local
 
-
-# The --glob setting excludes things that should not be linted, including
-# build artefacts, dependencies, and automatically generated files like
-# the changelogs (without excluding `changelogs/index.md`). Project proposals
-# are also excluded for want of a good way to ignore existing proposals and only
-# lint _new_ proposals. Project proposals aren't "living documents" in the way
-# the rest of our documentation is, so it doesn't seem right to retroactively
-# edit them for mere editorial purposes (rather than, for example, to correct
-# some grave inacurracy).
-# Leading `,` compensates for lack of trailing comma support
-# Note the lack of space before the trailing \ on each line, this is to prevent
-# the addition of a space between each pattern, which isn't supported by Vale's glob
-# If you change this and Vale takes a very long time to run (more than 30 seconds)
-# then chances are the change is breaking the glob pattern. Unfortunately the only
-# feedback you get when the glob pattern is not working as intended is a very long
-# run time for Vale as it checks everything that should have otherwise been ignored,
-# but wasn't due to some minor issue in the pattern syntax.
-# This is fiddly, but I can't find a good way around it.
-_IGNORE_PATTERNS := """
-    _build\
-    ,_static\
-    ,venv\
-    ,.venv\
-    ,.nuxt\
-    ,.pnpm-store\
-    ,node_modules\
-    ,test-results\
-    ,storebook-static\
-    ,.ruff-cache\
-    ,projects/proposals\
-    ,changelogs/api\
-    ,changelogs/frontend\
-    ,changelogs/catalog\
-    ,changelogs/ingestion_server\
-"""
-
-VALE_GLOB := "--glob='!*{" + _IGNORE_PATTERNS + "}*'"
-
+# 1. List all MD and MDX files in the repository
+# 2. Filter out the directories we do not want to lint
+# 3. Strip the absolute path from the beginning of the files
+# 4. Replace newlines with spaces
+FILES := ```
+    find $PWD/.. -type f \( -name "*.md" -o -name "*.mdx" \) \
+    | grep -vE ".pnpm|changelogs|projects/proposals|node_modules" \
+    | sed "s|$PWD/../||" \
+    | tr '\n' ' '
+```
+
+# Print the list of FILES for Vale to lint
+debug:
+    echo {{ FILES }} | tr ' ' '\n'
```

Comment:
Just a nit, will `tr ' ' '\n'` incorrect break files with legitimate spaces in their names? Not an issue for us now (and maybe never will be), but if `FILES` is a script now anyway, it can just be a recipe. The only reason it was a variable before was because it was easier to modify the list that way (I thought).

We can even break it into multiple steps to make it easier to read, and make the separator configurable (and conditional). It's a good opportunity to move the ignore pattern into a named space as well:

```
IGNORE_FILES_PATTERN := ".pnpm|changelogs|projects/proposals|node_modules|.?venv"

@files separator="\n":
    #! /usr/bin/env sh
    files=$(
        find "$PWD/.." -type f \( -name "*.md" -o -name "*.mdx" \) \
            | grep -vE "{{ IGNORE_FILES_PATTERN }}" \
            | sed "s|$PWD/../||"
    )
    if [ '{{ separator }}' != '\n' ]; then
        echo "$files" | tr '\n' '{{ separator }}'
        exit 0
    fi
    echo "$files"
```

I also somewhat doubt we should try to support passing a custom list of files to the run recipe, because now if I run `just .vale/run frontend`, it won't ignore the files it should ignore (and I won't find out until _after_ my PC has spent ages marching through the npm module tree linting every single README file).

Anyway, here is a diff that works with `run` able to take the `files` argument still (I just think we could do to remove it as well):

```diff
diff --git a/.vale/justfile b/.vale/justfile
index 4b9ad0043..9737b3c3f 100644
--- a/.vale/justfile
+++ b/.vale/justfile
@@ -11,27 +11,27 @@ NO_COLOR := "\\033[0m"
 build:
     docker build . -t openverse-vale:local
 
-# 1. List all MD and MDX files in the repository
-# 2. Filter out the directories we do not want to lint
-# 3. Strip the absolute path from the beginning of the files
-# 4. Replace newlines with spaces
-FILES := ```
-    find $PWD/.. -type f \( -name "*.md" -o -name "*.mdx" \) \
-    | grep -vE ".pnpm|changelogs|projects/proposals|node_modules" \
-    | sed "s|$PWD/../||" \
-    | tr '\n' ' '
-```
+IGNORE_FILES_PATTERN := ".pnpm|changelogs|projects/proposals|node_modules|.?venv"
 
-# Print the list of FILES for Vale to lint
-debug:
-    echo {{ FILES }} | tr ' ' '\n'
+files separator="\n":
+    #! /usr/bin/env sh
+    files=$(
+        find $PWD/.. -type f \( -name "*.md" -o -name "*.mdx" \) \
+            | grep -vE "{{ IGNORE_FILES_PATTERN }}" \
+            | sed "s|$PWD/../||"
+    )
+    if [ '{{ separator }}' != '\n' ]; then
+        echo "$files" | tr '\n' '{{ separator }}'
+        exit 0
+    fi
+    echo "$files"
 
 # Run Vale configured for Openverse in Docker.
 # Using Docker avoids the need for contributors to install the Vale binary.
 #Configuration defaults to what is used for CI.
-run +files=FILES: build
+run +files="": build
     docker run --rm \
         -v $PWD/..:/src:rw,Z \
         --workdir=/src \
         openverse-vale:local \
-        {{ files }}
+        {{ if files == "" { "`just files`" } else { files } }}
```

Reply:
@sarayourfriend this is exactly what I was looking for; thank you! The piece I was missing when trying to take this myself was making `files` a recipe, and generally thinking in terms of "composing recipes". 

I did decide to go ahead and remove allowing the user to pass their own files. I originally conceived of that being _specific_ file names and didn't consider the ignore issue when passing directories.

These changes are in cb69954.